File: go-server/handler/list.go
package handler

import (
	"github.com/gin-gonic/gin"
	"github.com/parikshitg/flutter_api_call/go-server/api"
)

func List(c *gin.Context) {
	res := api.ListResponse{}
	var users []api.User

	for _, v := range db {
		users = append(users, api.User{
			Name:  v.Name,
			Email: v.Email,
		})
	}

	res.Status.Success = true
	res.Users = users
	c.JSON(200, res)
}

File: flutter_client/lib/classes/list.dart
import './common.dart';

class User{
  final String name;
  final String email;

  User({this.name, this.email});

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      email: json['email'],
      name: json['name'],
    );
  }
}

class ListResponse{
  final Status status;
  final List<User> users;

  ListResponse({this.status, this.users});

  factory ListResponse.fromJson(Map<String, dynamic> json) {
    return ListResponse(
      status: Status.fromJson(json['status']),
      users: ((json['users']) as List).map((i) => User.fromJson(i)).toList(),
    );
  }
}